src/Screens/ChatScreen.tsx

```diff
@@ -81,15 +92,92 @@ export default function ChatScreen({route, navigation}: any) {
       });
   };
 
+  const isFocused = useIsFocused();
+
+  const fetchSingleDialog = () => {
+    setLoading(true);
+    console.log('dialogId', dialog?.id);
+    QB.chat
+      .getDialogs(dialog?.id)
+      .then((dialog: any) => {
+        time = dialog?.dialogs?.reduce(
+          (max: number, obj: any) =>
+            obj.customData.time > max ? obj.customData.time : max,
+          -Infinity,
+        );
+        cacluatetime();
+      })
+      .catch(error => {
+        console.error('Error retrieving dialog:', error);
+      });
+  };
+
+  useEffect(() => {
+    fetchSingleDialog();
+    return () => {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+        time = 0;
+      }
+    };
+  }, []);
+
+  const cacluatetime = () => {
+    if (isFocused) {
+      intervalRef.current = setInterval(() => {
+        if (time === 600 || time > 600) {
+          time = 600;
+          setChatBlocked(true);
+          clearInterval(intervalRef.current ? intervalRef.current : '');
+        }
+        time++;
+      }, 1000);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
you are removing the interval at line no 115 we you are doing again?

Reply:
sir in line 115, clear interval when screen geos out of focus.
And gain because if user is on chat screen and his time ends then I stopped the timer here and block the chat.